source/assets/scripts/filter.js

```diff
@@ -0,0 +1,369 @@
+export class Filter{
+  constructor(){}
+  /**
+   * Upon being called all recipe cards are filtered. From then on, 
+   * upon any checkbox being clicked, we refilter the collecting again.
+   */
+   filtering() {
+    //needed for filtering before new searches
+    this.#goThroughElements();
+
+    const checkboxByScore = document.querySelector("form[id=filter-by-score]");
+    checkboxByScore.addEventListener("click", () => {
+      // Could use this to add some more functionality to the all button.
+      this.#goThroughElements();
+    });
+
+    const checkboxByPrice = document.querySelector("form[id=filter-by-price]");
+    checkboxByPrice.addEventListener("click", () => {
+      this.#goThroughElements();
+    });
+
+    const checkboxByTime = document.querySelector("form[id=filter-by-time]");
+    checkboxByTime.addEventListener("click", () => {
+      this.#goThroughElements();
+    });
+  }
+
+  /**
+   * We iterate over the collection of recipe-cards and hide those that
+   * do meet the various filters.
+   * @private
+   */
+   #goThroughElements() {
```

Comment:
Hmm, why does this function need the # infront of the name? (i.e. why is it `#goThroughElements()` instead of `goThroughElements()`?)

Reply:
Is this somehow related to the fact that you want these functions to be `@private`?

Reply:
Right on. According to MDN, the convention of private class members/fields is to include a `#`. Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Thanks for linking the source too! ❤️